salt/proxy/dummy.py

```diff
@@ -0,0 +1,237 @@
+# -*- coding: utf-8 -*-
+'''
+This is a dummy proxy-minion designed for testing the proxy minion subsystem.
+'''
+from __future__ import absolute_import
+
+# Import python libs
+import os
+import logging
+import pickle
+
+# This must be present or the Salt loader won't load this module
+__proxyenabled__ = ['dummy']
+
+
+# Variables are scoped to this module so we can have persistent data
+# across calls to fns in here.
+DETAILS = {}
+
+DETAILS['services'] = {'apache': 'running', 'ntp': 'running', 'samba': 'stopped'}
+DETAILS['packages'] = {'coreutils': '1.0', 'apache': '2.4', 'tinc': '1.4', 'redbull': '999.99'}
+FILENAME = os.tmpnam()
+# Want logging!
+log = logging.getLogger(__file__)
+
+
+# This does nothing, it's here just as an example and to provide a log
+# entry when the module is loaded.
+def __virtual__():
+    '''
+    Only return if all the modules are available
+    '''
+    log.debug('dummy proxy __virtual__() called...')
+    return True
+
+
+def _save_state(details):
+    pck = open(FILENAME, 'wb')
+    pickle.dump(details, pck)
+    pck.close()
+
+
+def _load_state():
+    try:
+        pck = open(FILENAME, 'r')
+        DETAILS = pickle.load(pck)
+        pck.close()
+    except IOError:
+        DETAILS = {}
+        DETAILS['initialized'] = False
+        _save_state(DETAILS)
+
+    return DETAILS
+
+
+# Every proxy module needs an 'init', though you can
+# just put DETAILS['initialized'] = True here if nothing
+# else needs to be done.
+
+def init(opts):
+    log.debug('dummy proxy init() called...')
```

Comment:
Every proxymodule needs an init().  The dummy proxy doesn't do anything, though--there's nothing to initialize.